reflectable/build.yaml

```diff
@@ -4,6 +4,7 @@ builders:
     import: "package:reflectable/reflectable_builder.dart"
     builder_factories: ["reflectableBuilder"]
     build_extensions: {".dart": [".reflectable.dart"]}
+    required_inputs: [".dart"]
```

Comment:
@jakemac53 - double checking that you've had a chance to see this pattern and don't have any concerns with it.

Reflectable seems like it makes sense to both require and output .dart. Only 1 such builder can exist in a given project, but I think that is fine.

Reply:
I am a bit conflicted about it - but as a practical matter it makes sense for this package.

It does mean  you won't be able to use this package together with another builder which also has the same basic constraints though (but they might anyways be fundamentally incompatible).

Reply:
If it is a choice that usually works, could a client package (a package that depends on reflectable and runs the code generator) override that choice in their own build.yaml? In that case it could serve as a default setting that (presumably) almost always works.

Reply:
Otherwise, https://github.com/dart-lang/build/pull/3334 looks like the right thing to do.

Reply:
They can't easily override it (technically it is possible, you can override the entire build.yaml from package:reflectable, but it is not advised to do so and is a maintenance burden).

I am not opposed to this change in principal, it has both pros and cons, but I would probably just tell users to use the new feature coming in https://github.com/dart-lang/build/pull/3334 to avoid future issues.

Reply:
> It does mean you won't be able to use this package together with another builder which also has the same basic constraints though (but they might anyways be fundamentally incompatible).

I had the same concerns and ended up leaning towards the assumption that they'd be fundamentally incompatible anyway.

> I am not opposed to this change in principal, it has both pros and cons, but I would probably just tell users to use the new feature coming in [dart-lang/build#3334](https://github.com/dart-lang/build/pull/3334) to avoid future issues.

It's hard for me to get a sense of whether this current minor pain is likely to hit enough users to override the theoretical major pain if the incompatibility comes up. If we've only had this come up for a single user, then the extra config is the right tradeoff. Does anyone have any idea if there are many other users with the same problem?

Reply:
> I had the same concerns and ended up leaning towards the assumption that they'd be fundamentally incompatible anyway.

Fwiw, I don't actually think that is a fair assumption to make. The type of code that reflectable outputs is unlikely to be needed by other "global" builders. It could likely always run last.

Reply:
> It could likely always run last.

I'll keep that in mind if the issue comes up again.